src/sprite.c

```diff
@@ -345,7 +345,27 @@ s32 func_802DE5C8(s32 arg0) {
 
 INCLUDE_ASM(s32, "sprite", spr_free_sprite);
 
-INCLUDE_ASM(s32, "sprite", func_802DE748);
+typedef struct{
+    /* 0x00 */ char unk_00[0x4F];
+    /* 0x4F */ u8 unk_4F;
+} unk802DE748_2;
+
+typedef struct
+{
+    /* 0x00 */ unk802DE748_2** unk_00;
+    /* 0x04 */ char unk_04[0x10];
+} unk802DE748; // size = 0x14
+extern unk802DE748 D_802DFA4C[2];
+
+s32 func_802DE748(s32 arg0, s32 arg1) {
+    unk802DE748_2** temp_v1 = D_802DFA4C[arg0].unk_00;
```

Comment:
This function and the structs are super messy, unfortunately. It'd be good if you could spend some time cleaning this up a bit more and trying to figure out what these structs are. Also, the brace style for the second struct is wrong (there's an open brace on its own line) , and in the first struct, there's not a space between the word struct and the opening brace